site/content/en/reference/cfg/search/_index.md

```diff
@@ -0,0 +1,188 @@
+---
+title: "Search"
+linkTitle: "search"
+weight: 4
+type: docs
+description: >
+  Search and optionally replace fields across all resources.
+---
+
+<!--mdtogo:Short
+    Search and optionally replace fields across all resources.
+-->
+
+There is a spectrum of configuration customization techniques as described in
+[this document]. One of the most basic and simplest to understand is
+Search and Replace: The user fetches a package of configuration, searches all
+the files for fields matching a criteria, and replaces their values.
+
+Search matchers are provided by flags with `--by-` prefix. When multiple matchers
+are provided they are AND’ed together. `--put-` flags are mutually exclusive.
+
+### Synopsis
+
+<!--mdtogo:Long-->
+
+```
+kpt cfg search DIR [flags]
+```
+
+#### Args
+
+```
+DIR:
+  Path to a package directory
+```
+
+<!--mdtogo-->
+
+#### Flags
+
+```sh
+--by-value
+Match by value of a field.
+
+--by-value-regex
+Match by Regex for the value of a field. The syntax of the regular expressions
+accepted is the same general syntax used by Go, Perl, Python, and other languages.
+More precisely, it is the syntax accepted by RE2 and described at
+https://golang.org/s/re2syntax. With the exception that it matches the entire
+value of the field by default without requiring start (^) and end ($) characters.
+
+--by-path
+Match by path expression of a field. Path expressions are used to deeply navigate
+and match particular yaml nodes. Please note that path expressions are not
+regular expressions.
+
+--put-literal
+Set or update the value of the matching fields with the given literal value.
+
+--put-pattern
+Put the setter pattern as a line comment for matching fields.
+
+--recurse-subpackages
+Search recursively in all the nested subpackages.
+```
+
+### Examples
+
+<!--mdtogo:Examples-->
+
+```sh
+# Matches fields with value "3":
+$ kpt cfg search DIR/ --by-value 3
+```
+
+```sh
+# Matches fields with value prefixed by "nginx-":
+$ kpt cfg search DIR/ --by-value-regex ngnix-.*
```

Comment:
I had comments int the doc about quoting the values so it's not consumed by shell

Reply:
Thanks. Added single quotes.